lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1229,7 +1226,7 @@ struct StoreOpConversion
         asmArgs.emplace_back(llWord, constraint);
       }
 #ifdef USE_ROCM
-      store(asmArgs[vecStart].first, ptrElems[vecStart]);
```

Comment:
@rsanthanam-amd Again I want an explanation on why this change fixes stuffs. Looking at the logic essentially we are ditching the for loop of `vecStart`.

Reply:
There is a nested for-loop which populates the asmArgs data structure.  

This data structure is recreated for each vecStart loop iteration and for ROCm, the nWords loop condition is always 1 since we are only emitting a single store instruction per vecStart loop iteration.

My previous code incorrectly indexed into the asmArgs data structure with vecStart loop induction variable.

Reply:
Similar to my questions on load above:
- Can we be sure that 1 single load would suffice?
- What would be possible values of `nWords` for different data types?
- If we are sure `nWords` would always be 1, shall the loop which creates `asmArgs` be changed? Its original logic could be NV-specific where PTX inline asm may require the type of each argument be specified.
- In the earlier logic we change bitcast from `valArgTy` to `valueElemTy` between NV and AMD path, what are their values(types)? I guess `valueElemTy` could be a vector type, while `valArgTy` is always a scalar type? Is this understanding correct?
- Ultimately I feel we probably shouldn't depend on `asmArgs` vector anymore, as it no longer holds data we really care.

Reply:
I can't be sure that a single store would suffice but since we are not doing vectorized stores, it seems so.

nWords should always be 1 for scalar store.

"I guess valueElemTy could be a vector type, while valArgTy is always a scalar type" - I believe this is correct.

I will remove the use of the asmArgs vector for ROCm.

Reply:
pushed some changes which eliminate our dependence on asmArgs